engines/scalac/src/main/scala/scala/macros/internal/engines/scalac/trees/Abstracts.scala

```diff
@@ -193,126 +217,154 @@ trait Abstracts extends scala.macros.trees.Abstracts with Positions { self: Univ
           if (targs.nonEmpty) method = g.TypeApply(method, targs).setPos(op.pos)
           g.Apply(method, args)
         } else {
+          val termName = g.freshTermName()(g.currentFreshNameCreator)
```

Comment:
Expression: 1::2::Nil.
scalac generates:
`<synthetic> <artifact> val x$2: Int = 1;
{
  <synthetic> <artifact> val x$1: Int = 2;
  immutable.this.Nil.::(scala.Int.box(x$1))
}.::(scala.Int.box(x$2))`
macro: 
`val x$2: Int = 1;
{
  val x$1: Int = 2;
  immutable.this.Nil.::(scala.Int.box(x$1))
}.::(scala.Int.box(x$2))`
The difference is macro does not show '<synthetic> <artifact>' for val. 
@xeno-by is it important to show it?